### PR TITLE
Update Developer's Guide

### PR DESCRIPTION
**Pull request by potty10**
Update the following sections in the developer's guide:
- Acknowledgements
- Non functional requirements
- Glossary

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -369,16 +371,16 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+3.  An undergraduate student with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
-
+* **Computing undergraduate**: Refers to an undergraduate of the National University of Singapore who is currently studying towards a computing degree.
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:382]: Why specifically NUS? Would it suffice to mention _Refers to a university undergraduate pursuing a computing degree_?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -9,7 +9,9 @@ title: Developer Guide
 
 ## **Acknowledgements**
 
-* {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* InternBuddy is writtin in **Java 11**, with usage of the [JavaFX](https://openjfx.io/),
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:12]: `written` and not `writtin`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -369,16 +371,16 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+2.  Should be able to hold up to 1000 internship entries without a noticeable sluggishness in performance for typical usage.
+3.  An undergraduate student with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:375]: Is there a reason why Computing is omitted in this case? i.e. `undergraduate` instead of `Computing undergraduate`.

**Review by eugenetangkj — COMMENTED** [review]
Good effort, some questions on differences between undergraduate and Computing undergraduate.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (potty10) [reply docs/DeveloperGuide.md]: Noted!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (potty10) [reply docs/DeveloperGuide.md]: I agree.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (potty10) [reply docs/DeveloperGuide.md]: I agree it should be consistent.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DerrickSaltFish) [reply docs/DeveloperGuide.md]: Agree

**Comment by eugenetangkj** [comment]
Looks good to merge.

**Pull request merged**